### PR TITLE
Upload place story

### PR DESCRIPTION
**Pull request by talbenasuli**

### CODE REVIEW

File: Dizzy/Screens/SignIn/SignInWithDizzyVC.swift [hunk 1]
```diff
@@ -78,7 +78,8 @@ extension SignInWithDizzyVC: SignInDetailsViewDelegate {
 extension SignInWithDizzyVC: SignInWithDizzyVMDelegate {
     func validationFailed(inputValidation: InputValidationResult) {
         self.hideSpinner()
-        showAlert(title: "Validation Error".localized, message: inputValidation.rawValue)
+//        showPopup
```
Comment (menesh2) [inline Dizzy/Screens/SignIn/SignInWithDizzyVC.swift]: remove marks

File: Dizzy/Screens/SignIn/SignInWithDizzyVC.swift [hunk 2]
```diff
@@ -87,6 +88,6 @@ extension SignInWithDizzyVC: SignInWithDizzyVMDelegate {
     
     func userSignedInFailed(error: SignInWebserviceError) {
         self.hideSpinner()
-        showAlert(title: "Error".localized, message: error.localizedDescription)
+//        showAlert(title: "Error".localized, message: error.localizedDescription)
```
Comment (menesh2) [inline Dizzy/Screens/SignIn/SignInWithDizzyVC.swift]: remove marks

File: Dizzy/Screens/UploadStory/PhotoPresenterVM.swift [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+}
+
+final class PhotoPresenterVM: PhotoPresenterVMType {
+    
+    var photo: UIImage
+    let uploadFileInteractor: UploadFileInteractorType
+    let placeInfo: PlaceInfo
+    var loading = Observable<Bool>(false)
+    
+    var delegate: PhotoPresenterVMDelegate?
```
Comment (menesh2) [inline Dizzy/Screens/UploadStory/PhotoPresenterVM.swift]: use weak var

File: Dizzy/Screens/UploadStory/UploadStoryCoordinator.swift [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    var onCoordinatorFinished: (_ dismiss: Bool) -> Void { get set }
+}
+
+final class UploadStoryCoordinator: UploadStoryCoordinatorType {
+    
+    var onCoordinatorFinished: (_ dismiss: Bool) -> Void = { _ in }
+    var navigationController: UINavigationController
+    var container: Container?
+    var childCoordinators: [CoordinatorKey : Coordinator] = [:]
+    let placeInfo: PlaceInfo
```
Comment (menesh2) [inline Dizzy/Screens/UploadStory/UploadStoryCoordinator.swift]: coordinator does not suppose to hold any data, you can get it with in the delegate call

File: Dizzy/Screens/UploadStory/UploadStoryVC.swift [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+        cameraButton.backgroundColor = .clear
+        cameraButton.layer.borderColor = UIColor.white.cgColor
+        cameraButton.layer.borderWidth = borderRodius
+        view.layoutIfNeeded()
+        cameraButton.layer.cornerRadius = cameraButton.frame.height / 2
+        cameraButton.addTarget(self, action: #selector(cameraButtonPressed), for: .touchUpInside)
+    }
+    
+    private func bindViewModel() {
+        viewModel.captureVideoLayer.bind {                   [weak self] layer in
```
Comment (menesh2) [inline Dizzy/Screens/UploadStory/UploadStoryVC.swift]: please arrange spaces

File: Dizzy/UI Components/PopupPresenter.swift [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+//  Dizzy
+//
+//  Created by Tal Ben Asuli on 19/06/2019.
+//  Copyright © 2019 Dizzy. All rights reserved.
+//
+
+import UIKit
+
+enum PopupButtonLayer {
+    case oneButton(buttonText: String, onClick: (() -> Void)?)
```
Comment (menesh2) [inline Dizzy/UI Components/PopupPresenter.swift]: why wont the user get array of actions and the you could create any alert with any actions in one func

File: Dizzy/Web Service/FirebaseWebService.swift [hunk 7]
```diff
@@ -56,15 +56,39 @@ final class FirebaseWebService: WebServiceType {
...
                 }
                 
             }
         }
     }
     
+    func uplaodFile(with path: String, data: UploadFileData,  completion: @escaping (Result<UploadFileResponse>) -> Void) {
+        let ref = storageReference.child(path)
+        let uploadTask = ref.putData(data.data, metadata: nil) { (metaData, error) in
+            if let error = error {
```
Comment (menesh2) [inline Dizzy/Web Service/FirebaseWebService.swift:70]: move this code into separate func

**Review by menesh2 — CHANGES_REQUESTED** [review]
you have some swiftLint errors, please fix them

File: Dizzy/Screens/SignIn/SignInWithDizzyVC.swift (hunk 1 above)
Comment (talbenasuli) [reply Dizzy/Screens/SignIn/SignInWithDizzyVC.swift]: Done

File: Dizzy/Screens/SignIn/SignInWithDizzyVC.swift (hunk 2 above)
Comment (talbenasuli) [reply Dizzy/Screens/SignIn/SignInWithDizzyVC.swift]: Done

File: Dizzy/Screens/UploadStory/PhotoPresenterVM.swift (hunk 3 above)
Comment (talbenasuli) [reply Dizzy/Screens/UploadStory/PhotoPresenterVM.swift]: Done

File: Dizzy/Screens/UploadStory/UploadStoryVC.swift (hunk 5 above)
Comment (talbenasuli) [reply Dizzy/Screens/UploadStory/UploadStoryVC.swift]: oops, done

File: Dizzy/Web Service/FirebaseWebService.swift (hunk 7 above)
Comment (talbenasuli) [reply Dizzy/Web Service/FirebaseWebService.swift:70]: Done

File: Dizzy/UI Components/PopupPresenter.swift (hunk 6 above)
Comment (talbenasuli) [reply Dizzy/UI Components/PopupPresenter.swift]: Done

File: Dizzy/Screens/UploadStory/UploadStoryCoordinator.swift (hunk 4 above)
Comment (talbenasuli) [reply Dizzy/Screens/UploadStory/UploadStoryCoordinator.swift]: Done.

**Review by menesh2 — APPROVED** [review]

**Pull request merged**